src/testing/TestFields.tsx

```diff
@@ -0,0 +1,151 @@
+import React, { FC } from 'react';
+import { Box, Text, Label, Input, Checkbox, Textarea, Radio } from 'theme-ui';
+import { FormikContextType, FormikValues } from 'formik';
+import { Message } from 'retranslate';
+import * as yup from 'yup';
+
+import { TestType, ObjectSchema, FieldSchema } from '../api';
+
+interface TestFieldsProps {
+  form: FormikContextType<FormikValues>;
+  testType: TestType;
+}
+
+export const TestFields: FC<TestFieldsProps> = ({ form, testType }) => {
+  const fieldError = (key: keyof TestType['resultsSchema']['properties']) =>
+    form.touched[key] && form.errors[key] ? <Text>{form.errors[key]}</Text> : null;
+
+  return (
+    <>
+      {Object.entries(testType.resultsSchema.properties).map(([key, value]) => {
+        if (value.oneOf) {
+          return (
+            <Box key={key}>
+              {value.title} {testType.resultsSchema.required?.includes(key) && '*'}
+              {value.oneOf.map((option) => (
+                <Label key={`${option.const}`}>
+                  <Radio
+                    checked={form.getFieldProps(key).value === option.const}
+                    value={typeof option.const === 'number' ? option.const : `${option.const}`}
```

Comment:
Same question here, does the value need to be a string?

Reply:
Yup, that's the tricky part — it needs to be a number or a string, so for boolean, we need to cast it to a string and also convert back to a boolean in the change handler.